superset/assets/javascripts/reduxUtils.js

```diff
@@ -95,5 +96,5 @@ export function areArraysShallowEqual(arr1, arr2) {
 }
 
 export function areObjectsEqual(obj1, obj2) {
-  return JSON.stringify(obj1) === JSON.stringify(obj2);
+  return equals(obj1, obj2);
```

Comment:
This will compare leaves using `==` instead of `===`, is that the intended behavior? If not, you can pass `opts.strict` as `true`.

Reply:
Good catch. `==` would have been ok in most cases but `===` is preferable. We may not need this utility function anymore but I'm leaving it to minimize code changes.